contracts/zero-ex/tests/utils/MultiplexUtils.sol

```diff
@@ -0,0 +1,304 @@
+        address[] memory tokens,
+        uint256 sellAmount
+    ) internal pure returns (IMultiplexFeature.BatchSellSubcall memory) {
+        return
+            IMultiplexFeature.BatchSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.UniswapV3,
+                sellAmount: sellAmount,
+                data: _encodePathUniswapV3(tokens)
+            });
+    }
+
+    function _makeMockLiquidityProviderMultiHopSubcall()
+        internal
+        view
+        returns (IMultiplexFeature.MultiHopSellSubcall memory)
+    {
+        return
+            IMultiplexFeature.MultiHopSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.LiquidityProvider,
+                data: abi.encode(address(liquidityProvider), hex"")
+            });
+    }
+
+    function _makeMockLiquidityProviderBatchSubcall(
+        uint256 sellAmount
+    ) internal view returns (IMultiplexFeature.BatchSellSubcall memory) {
+        return
+            IMultiplexFeature.BatchSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.LiquidityProvider,
+                sellAmount: sellAmount,
+                data: abi.encode(address(liquidityProvider), hex"")
+            });
+    }
+
+    function _makeMockTransformERC20Subcall(
+        IERC20TokenV06 inputToken,
+        IERC20TokenV06 outputToken,
+        uint256 sellAmount,
+        uint256 mintAmount
+    ) internal view returns (IMultiplexFeature.BatchSellSubcall memory) {
+        ITransformERC20Feature.Transformation[] memory transformations = new ITransformERC20Feature.Transformation[](1);
+        transformations[0] = ITransformERC20Feature.Transformation(
+            uint32(transformerNonce),
+            abi.encode(address(inputToken), address(outputToken), sellAmount, mintAmount, 0)
+        );
+
+        return
+            IMultiplexFeature.BatchSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.TransformERC20,
+                sellAmount: sellAmount,
+                data: abi.encode(transformations)
+            });
+    }
+
+    function _makeNestedBatchSellSubcall(
+        IMultiplexFeature.BatchSellSubcall[] memory calls
+    ) internal pure returns (IMultiplexFeature.MultiHopSellSubcall memory) {
+        return
+            IMultiplexFeature.MultiHopSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.BatchSell,
+                data: abi.encode(calls)
+            });
+    }
+
+    function _makeNestedMultiHopSellSubcall(
+        address[] memory tokens,
+        IMultiplexFeature.MultiHopSellSubcall[] memory calls,
+        uint256 sellAmount
+    ) internal pure returns (IMultiplexFeature.BatchSellSubcall memory) {
+        return
+            IMultiplexFeature.BatchSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.MultiHopSell,
+                sellAmount: sellAmount,
+                data: abi.encode(tokens, calls)
+            });
+    }
+
+    function _makeArray(
```

Comment:
It seems a little strange to have to define so many different versions of makeArray, that are basically all the same but can have different types and different numbers of arguments. I assume there's no inline way to do this in solidity. You do use this in almost 50 places, so I can see that it is saving some lines of code to have these functions. I just wonder if there's a better / cleaner and more general way to do this.